### DISCUSSION TITLE
java.compile.nullAnalysis.nonnull seems to raise false positives

### DISCUSSION THREAD

**Discussion by lebarjack** [redhat-developer/vscode-java#3507] [Q&A] [answered]
Hi, I am a little confused with recent NullAnalysis.
My spring-boot project now raise this warning:
`Null type safety: The expression of type 'String' needs unchecked conversion to conform to '@NonNull String'`

On the following code:
`

    @NonNull
    String returnString(@NonNull String myString) {
        StringBuilder urlBuilder = new StringBuilder(myString);
        return urlBuilder.toString();
    }

    @NonNull
    String returnStringWithCheck(@NonNull String myString) {
        StringBuilder urlBuilder = new StringBuilder(myString);
        if (null != urlBuilder.toString())
            return urlBuilder.toString();
        else
            return "";
    }
`
It's raised on the (lombok) getters too:

`

    @NonNull
    String returnLombokGetter(@NonNull Account account) {
        if (null != account.getDisplayName())
            return account.getDisplayName();
        else
            return "";
    }`
   
NB: I use the following NonNull: org.springframework.lang.NonNull
And null analysis mode is on 'interactive'
I restrained the NonNull analysis like this:
`"java.compile.nullAnalysis.nonnull": ["org.springframework.lang.NonNull"]`

Some complementary informations:

- Operating System: bookworm container with openjdk 21
- Visual Studio Code version: 1.86.1
- Java extension version: 1.28.1
- Spring Boot 3.2.2

Is this a bug or I don't do the right thing?

**Comment by snjeza** [comment, marked as answer]
@lebarjack You may want to take a look at https://github.com/redhat-developer/vscode-java/issues/3501#issuecomment-1952529229
You can also try the pre-release version of VS Code Java .

**Comment by lebarjack** [comment]
@snjeza  It might be related, et I had no jakarta constraint in my tests.
I occurs while using exclusively spring NonNull annotations.

I am not familiar with building the extension myself.
Is there an easy way to check with the pre-release?

**Reply by lebarjack** [reply]
Oh sorry, I found it directly in the extension repository in vscode.

**Reply by lebarjack** [reply]
The pre-release version doesn't solve the problem.

**Comment by snjeza** [comment]
@lebarjack Could you show `<your_project/.settings/org.eclipse.jdt.core.prefs`

**Reply by lebarjack** [reply]
Hi, 
This morning, I have rebuilt my container, switched to prerelease and reloaded. It seems to work as it should.
Thanks a lot for your help.
